trial/src/Trial.hs

```diff
@@ -9,7 +9,24 @@ Copyright: (c) 2020 Kowainik
 SPDX-License-Identifier: MPL-2.0
 Maintainer: Kowainik <xrom.xkov@gmail.com>
 
-Trial Data Type
+'Trial' data structure is 'Either'-like structure that stores tagged
+history of all events. 'Trial' has two constructors:
+
+* 'Fiasco': stores the list of tagged events, at least one event has
```

Comment:
I guess you put a different semantics into `tag` word, which is a bit confusing for me. I don't now see the difference between `Fatality` and `TaggedTrial` which is not good. Do you want to rename the data types then? I will update my documentation properly.
What would you use for `TaggedTrial` then? `SourcedTrial`?

Reply:
@vrom911 No, I just forgot the word `level` while I was looking on how to name the values of type `Fatality`. And I understand that this can be confusing, especially when we distinguish tags and provide a separate use-case for it, I can see this now. I will rewrite relevant parts to be more clear.